packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/model/FlutterSerializedModel.kt

```diff
@@ -51,34 +51,38 @@ data class FlutterSerializedModel(val serializedModel: SerializedModel) {
         )
 
         return serializedData.mapValues {
-            val field = modelSchema.fields[it.key]!!
-            when (val value: Any = it.value) {
-                is Temporal.DateTime -> value.format()
-                is Temporal.Date -> value.format()
-                is Temporal.Time -> value.format()
-                is Model -> FlutterSerializedModel(value as SerializedModel).toMap()
-                is Temporal.Timestamp -> value.secondsSinceEpoch
-                is SerializedCustomType -> FlutterSerializedCustomType(value).toMap()
-                is List<*> -> {
-                    if (field.isCustomType) {
-                        // for a list like field if its type is CustomType
-                        // Then the item type must be CustomType
-                        (value as List<SerializedCustomType>).map { item ->
-                            FlutterSerializedCustomType(item).toMap()
+            val field = modelSchema.fields[it.key]
+            if (field == null) {
+                null
+            } else {
+                when (val value: Any = it.value) {
+                    is Temporal.DateTime -> value.format()
+                    is Temporal.Date -> value.format()
+                    is Temporal.Time -> value.format()
+                    is Model -> FlutterSerializedModel(value as SerializedModel).toMap()
+                    is Temporal.Timestamp -> value.secondsSinceEpoch
+                    is SerializedCustomType -> FlutterSerializedCustomType(value).toMap()
+                    is List<*> -> {
+                        if (field.isCustomType) {
+                            // for a list like field if its type is CustomType
+                            // Then the item type must be CustomType
+                            (value as List<SerializedCustomType>).map { item ->
```

Comment:
```suggestion
                            (value.cast<SerializedCustomType>()).map { item ->
```

Reply:
This breaks for some cases when the value uses the `List` implementation is a variant other than `ArrayList`. E.g. `SingletonList` created by Kotlin `listOf`.

I think the `cast` function should keep the generic `List`?

Reply:
The definition of `cast` is incorrect... I'm not sure why because I had written it differently at one point. It should be

```kotlin
inline fun <reified T> List<*>.cast(): List<T> =
    this as List<T>
```

Reply:
I've updated the `cast` function with the latest commit, thanks.